tests/gateway/providers/test_mosaicml.py

```diff
@@ -0,0 +1,362 @@
+                "mosaicml_api_key": "key",
+            },
+        },
+    }
+
+
+def embeddings_response():
+    return {
+        "outputs": [
+            [
+                3.25,
+                0.7685547,
+                2.65625,
+                -0.30126953,
+                -2.3554688,
+                1.2597656,
+            ]
+        ],
+        "headers": {"Content-Type": "application/json"},
+    }
+
+
+@pytest.mark.asyncio
+async def test_embeddings():
+    resp = embeddings_response()
+    config = embeddings_config()
+    with mock.patch(
+        "aiohttp.ClientSession.post", return_value=MockAsyncResponse(resp)
+    ) as mock_post:
+        provider = MosaicMLProvider(RouteConfig(**config))
+        payload = {"text": "This is a test"}
+        response = await provider.embeddings(embeddings.RequestPayload(**payload))
+        assert jsonable_encoder(response) == {
+            "embeddings": [
+                [
+                    3.25,
+                    0.7685547,
+                    2.65625,
+                    -0.30126953,
+                    -2.3554688,
+                    1.2597656,
+                ]
+            ],
+            "metadata": {
+                "input_tokens": None,
+                "output_tokens": None,
+                "total_tokens": None,
+                "model": "instructor-large",
+                "route_type": "llm/v1/embeddings",
+            },
+        }
+        mock_post.assert_called_once()
+
+
+@pytest.mark.asyncio
+async def test_param_model_is_not_permitted():
+    config = embeddings_config()
+    provider = MosaicMLProvider(RouteConfig(**config))
+    payload = {
+        "prompt": "This should fail",
+        "max_tokens": 5000,
+        "model": "something-else",
+    }
+    with pytest.raises(HTTPException, match=r".*") as e:
+        await provider.completions(completions.RequestPayload(**payload))
+    assert "The parameter 'model' is not permitted" in e.value.detail
+    assert e.value.status_code == 422
+
+
+@pytest.mark.parametrize("prompt", [{"set1", "set2"}, ["list1"], [1], ["list1", "list2"], [1, 2]])
+@pytest.mark.asyncio
+async def test_completions_throws_if_prompt_contains_non_string(prompt):
+    config = completions_config()
+    provider = MosaicMLProvider(RouteConfig(**config))
+    payload = {"prompt": prompt}
+    with pytest.raises(ValidationError, match=r"prompt"):
+        await provider.completions(completions.RequestPayload(**payload))
+
+
+@pytest.mark.parametrize(
+    ("messages", "expected_output"),
+    [
+        (
+            [
+                RequestMessage(role="system", content="Hello"),
+                RequestMessage(role="user", content="Hi there"),
+                RequestMessage(role="assistant", content="How can I help?"),
+                RequestMessage(role="user", content="Thanks!"),
+            ],
+            "<s>[INST] <<SYS>> Hello <</SYS>> Hi there [/INST] How can I help? </s>"
+            "<s>[INST] Thanks! [/INST]",
```

Comment:
We need closing `</s>` here and in the other tests, right @arpitjasa-db ?

Reply:
No actually, we're supposed to leave it open for the final instruction
<img width="892" alt="Screenshot 2023-08-29 at 3 08 48 PM" src="https://github.com/mlflow/mlflow/assets/87999496/ea1e4f51-c425-4d52-9e6f-b8cb6cd10acb">
<img width="885" alt="Screenshot 2023-08-29 at 3 08 54 PM" src="https://github.com/mlflow/mlflow/assets/87999496/44887370-0458-40ea-b428-c3428c5e82fc">

Reply:
Got it!